pkg/backends/newrelic/newrelic.go

```diff
@@ -294,24 +311,29 @@ func (n *Client) constructPost(ctx context.Context, buffer *bytes.Buffer, data i
 		return nil, fmt.Errorf("[%s] unable to marshal: %v", BackendName, mErr)
 	}
 
-	return n.postWrapper(ctx, mJSON)
+	return n.postWrapper(ctx, mJSON, "metrics")
 }
 
 // postWrapper compresses JSON for Insights
-func (n *Client) postWrapper(ctx context.Context, json []byte) (func() error, error) {
+func (n *Client) postWrapper(ctx context.Context, json []byte, dataType string) (func() error, error) {
 	return func() error {
 		headers := map[string]string{
 			"Content-Type": "application/json",
 			"User-Agent":   n.userAgent,
 		}
 
 		// Insights Event API requires gzip or deflate compression
-		if n.flushType == flushTypeInsights && n.apiKey != "" {
+		// https://docs.newrelic.com/docs/insights/insights-data-sources/custom-data/introduction-event-api#h2-basic-workflow
+		// Metrics API requires gzip or identity
```

Comment:
Oh man, I'm going to have to add gzip to my http client refactoring!

Should be pretty trivial though :)

Reply:
So these need to be kept separate, the reason for that is our new dimensional metrics pipeline cannot accept arbitrary events on the same endpoint. 

When you use address-metrics and supply it the new metrics endpoint it will ship gostatsd metrics as NR dimensional metrics. However if you want to also send gostatsd events, you will need to configure our insights collector on that original address option as well.

We've documented this clearly in the BACKENDS.md.
Unfortunately at this moment there is no easier way to simplify this right now.